### ISSUE TITLE
PHP 7.4 Incompatibility: Covariant return / contravariant parameters are reported as errors

### ISSUE THREAD

**Issue by tux-rampage** [vimeo/psalm#2264]
PHP 7.4 implements [Covariant Returns and Contravariant Parameters](https://wiki.php.net/rfc/covariant-returns-and-contravariant-parameters#php_rfccovariant_returns_and_contravariant_parameters).

So this code becomes valid in PHP 7.4, but will error in psalm:

```php
<?php

class A 
{
  public function bar(Iterator $items): void
  {
  }
  
  public static function foo(): self
  {
    return new self();
  }
}

// This is valid in PHP 7.4: https://wiki.php.net/rfc/covariant-returns-and-contravariant-parameters
class B extends A {
  public function bar(Traversable $items): void
  {
  }

  public static function foo(): self
  {
    return new self();
  }
}
```

Example: https://psalm.dev/r/65e66d589f

**Comment by weirdan** [comment]
Partial duplicate (where covariant returns are concerned) of #2102
